src/injected-command-handler.ts

```diff
@@ -0,0 +1,456 @@
+    if (node) {
+      this.clickNode(node);
+    }
+  }
+
+  async COMMAND_TYPE_DOM_COPY(data: any): Promise<any> {
+    const element = document.querySelector(data.text) as HTMLElement;
+    if (element && window.getSelection) {
+      let selection = window.getSelection();
+      let range = document.createRange();
+      range.selectNodeContents(element);
+      selection?.removeAllRanges();
+      selection?.addRange(range);
+      await navigator.clipboard.writeText(selection?.toString() || "");
+      selection?.removeAllRanges();
+    }
+  }
+
+  async COMMAND_TYPE_DOM_FOCUS(data: any): Promise<any> {
+    const element = document.querySelector(data.text) as HTMLElement;
+    if (element) {
+      element.focus();
+    }
+  }
+
+  async COMMAND_TYPE_DOM_SCROLL(data: any): Promise<any> {
+    const element = document.querySelector(data.text) as HTMLElement;
+    if (element) {
+      element.scrollIntoView({
+        block: "center",
+        inline: "center",
+        behavior: "smooth",
+      });
+    }
+  }
+
+  async COMMAND_TYPE_GET_EDITOR_STATE(_data: any): Promise<any> {
+    const editor = await editors.active();
+    if (!editor) {
+      return;
+    }
+    return editor.getEditorState();
+  }
+
+  async COMMAND_TYPE_REDO(_data: any): Promise<any> {
+    const editor = await editors.active();
+    editor?.redo();
+    return;
+  }
+
+  async COMMAND_TYPE_SCROLL(data: any): Promise<any> {
+    if (data.direction) {
+      this.scrollInDirection(data.direction);
+    }
+    if (data.path) {
+      if (data.path === "top" || data.path === "bottom") {
+        return this.scrollInDirection(data.path);
+      } else {
+        return this.findAndScroll(data.path);
+      }
+    }
+    return;
+  }
+
+  async COMMAND_TYPE_SELECT(data: any): Promise<any> {
+    const editor = await editors.active();
+    if (!editor) {
+      return;
+    }
+
+    return editor.setSelection(data.cursor, data.cursorEnd);
+  }
+
+  async COMMAND_TYPE_SHOW(data: any): Promise<any> {
+    let selector = "";
+    if (data.text == "links") {
+      selector = 'a, button, summary, [role="link"], [role="button"]';
+    } else if (data.text == "inputs") {
+      selector =
+        'input, textarea, [role="checkbox"], [role="radio"], .CodeMirror';
+    } else if (data.text == "code") {
+      selector = "pre, code";
+    } else {
+      return;
+    }
+    const nodes = this.nodesMatchingSelector(selector);
+    this.showOverlays(Array.from(nodes), data.text);
+  }
+
+  async COMMAND_TYPE_UNDO(_data: any): Promise<any> {
+    const editor = await editors.active();
+    editor?.undo();
+    return;
+  }
+
+  async COMMAND_TYPE_USE(data: any): Promise<any> {
+    let overlay = this.overlays[data.index - 1];
+    if (overlay.type === "links" || overlay.type === "inputs") {
+      this.clickNode(overlay.node);
+    } else if (overlay.type === "code") {
+      await this.copyCode(overlay.node);
+      alert(`Copied ${data.index}`);
```

Comment:
we should avoid using `alert`—it's intrusive and is going to be removed from Chrome soon anyway

Reply:
Replaced this with an overlay div in the center of the screen that times out after a second.